src/Build/Graph/ProjectGraph.cs

```diff
@@ -56,6 +57,8 @@ public sealed class ProjectGraph
 
         private readonly Lazy<IReadOnlyCollection<ProjectGraphNode>> _projectNodesTopologicallySorted;
 
+        private readonly EvaluationContext _evaluationContext = EvaluationContext.Create(EvaluationContext.SharingPolicy.Shared);
```

Comment:
Unsure if it's worth it: `static`?

Reply:
Nvm, wouldn't want to share across different graphs, which might be constructed at different times.